### DISCUSSION TITLE
Creating Custom EventType possible?

### DISCUSSION THREAD

**Discussion by tobimuell** [keycloak/keycloak#15213] [Q&A]
Hi there,

i would like to combine two keycloak extensions loosely coupled via a custom EventType. In org.keycloak.events.* there is an Enum called EventType which is used to emit events. These events can be catched with an EventListenerProvider. So far so good. 

- How can i extend these EventType? 

- Is there another solution to emit custom events? 

- If we use a existing EventType e.g. CLIENT_INFO and add custom details can we expect side effects of other keycloak components who works with it or is that case as an alternativ ok?

Thanks for your replies!

**Comment by jbman** [comment]
It would be very helpful if Keycloak extensions could define custom events and emit those events. We have extensions which send out additional emails (e.g an invitation email). We would use it to create an event when an invitation email was sent by this extension.

Related ticket by original author @tobimuell : https://github.com/keycloak/keycloak/issues/15288
